federatedgroups/lib/Controller/ScimController.php

```diff
@@ -51,275 +41,254 @@ function getOurDomain() {
+        $newMembers     = [];
+        foreach ($obj["members"] as $member) {
+            $userIdParts = explode("@", $member["value"]);
+            if (count($userIdParts) == 3) {
+                $userIdParts = [$userIdParts[0] . "@" . $userIdParts[1], $userIdParts[2]];
+            }
+            if (count($userIdParts) != 2) {
+                throw new Exception("cannot parse OCM user " . $member["value"]);
+            }
+            $newMember = $userIdParts[0];
+            if ($userIdParts[1] !== getOurDomain()) {
+                $newMember .= "#" . $userIdParts[1];
+            }
+            $newMembers[] = $newMember;
+        }
+        foreach ($currentMembers as $currentMember) {
+            if (!in_array($currentMember, $newMembers)) {
+                $backend->removeFromGroup($currentMember, $groupId);
+            }
+        }
+
+        foreach ($newMembers as $newMember) {
+            if (!in_array($newMember, $currentMembers)) {
+                $backend->addToGroup($newMember, $groupId);
+
+                $newDomain = $this->getNewDomainIfNeeded($newMember, $currentMembers);
+                if ($newDomain) {
+                    $this->mixedGroupShareProvider->sendOcmInviteForExistingShares($newDomain, $groupId);
+                }
+            }
+        }
+    }
+
+    /**
+     * @NoCSRFRequired
+     * @PublicPage
+     */
+    public function createGroup($id, $members) {
+        if (!$id) {
+            return new JSONResponse(['status' => 'Error', 'message' => "Missing id param"], Http::STATUS_BAD_REQUEST);
+        } else if (!$members) {
+            return new JSONResponse(['status' => 'Error', 'message' => "Missing members param"], Http::STATUS_BAD_REQUEST);
+        }
+        
+        $body = ["id" => $id, "members" => $members];
+
+
+        $this->groupManager->createGroup($id);
+
+        // expect group to already exist
+        // we are probably receiving this create due to
+        // https://github.com/SURFnet/rd-sram-integration/commit/38c6289fd85a92b7fce5d4fbc9ea3170c5eed5d5
+        $this->handleUpdateGroup($id, $body);
+        return new JSONResponse(
+            $body,
+            Http::STATUS_CREATED
+        );
+    }
+
+    /**
+     * @NoCSRFRequired
+     * @PublicPage
+     */
+    public function updateGroup($groupId, $members) {
+        if (!$groupId) {
+            return new JSONResponse(['status' => 'Error', 'message' => "Missing groupId param"], Http::STATUS_BAD_REQUEST);
+        } else if (!$members) {
+            return new JSONResponse(['status' => 'Error', 'message' => "Missing members param"], Http::STATUS_BAD_REQUEST);
+        }
+
+        $body = ["members" => $members];
+
+        $this->handleUpdateGroup($groupId, $body);
+        return new JSONResponse(
+            $body,
+            Http::STATUS_OK
+        );
+    }
+
+    /**
+     * @NoCSRFRequired
+     * @PublicPage
+     */
+    public function deleteGroup($groupId) {
```

Comment:
should groupId checked too?

Reply:
no, because groupId in the case of deleteGroup is part of the URL, and not provided it should be resulted in a 404(breaking the rest)